diff --git a/worker/imap/configure.go b/worker/imap/configure.go
index 32b0c7be..57b80b12 100644
--- a/worker/imap/configure.go
+++ b/worker/imap/configure.go
@@ -8,6 +8,7 @@ import (
 	"strconv"
 	"strings"
 	"time"
+	"path"
 
 	"git.sr.ht/~rjarry/aerc/lib/xdg"
 	"git.sr.ht/~rjarry/aerc/worker/lib"
@@ -178,19 +179,17 @@ func (w *IMAPWorker) handleConfigure(msg *types.Configure) error {
 	w.idler = newIdler(w.config, w.worker, w.executeIdle)
 	w.observer = newObserver(w.config, w.worker)
 
-	if name, ok := msg.Config.Params["folder-map"]; ok {
-		file := xdg.ExpandHome(name)
-		f, err := os.Open(file)
-		if err != nil {
-			return err
-		}
-		defer f.Close()
-		fmap, order, err := lib.ParseFolderMap(bufio.NewReader(f))
-		if err != nil {
-			return err
-		}
-		w.worker = middleware.NewFolderMapper(w.worker, fmap, order)
+	file := path.Join(xdg.ConfigPath("aerc"), "folder-map")
+	f, err := os.Open(file)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	fmap, order, err := lib.ParseFolderMap(bufio.NewReader(f))
+	if err != nil {
+		return err
 	}
+	w.worker = middleware.NewFolderMapper(w.worker, fmap, order)
 
 	return nil
 }
